### ISSUE TITLE
🐛 Bug Report: Should have renderNode and renderLabel prop to CatalogGraphPage

### ISSUE THREAD

**Issue by shubhamjajoo** [backstage/backstage#17096]
### 📜 Description

**CatalogGraphPage** does not have option to pass ```renderNode``` and ```renderLabel``` as props.
So not able to render custom node and label on ```/catalog-graph``` route.

As the child component have those options to render custom node and label but CatalogGraphPage is not accepting those props.

### 👍 Expected behavior

It should accept ```renderNode``` and ```renderLabel``` prop and pass it down to ```EntityRelationsGraph``` component, this is what I feel.

Please check this draft [PR](https://github.com/backstage/backstage/pull/17097)

### 👎 Actual Behavior with Screenshots

**CatalogGraphPage**
<img width="842" alt="image" src="https://user-images.githubusercontent.com/12490835/227731686-14fd39fc-36c0-479c-b467-cf1ac5754e21.png">

### 👟 Reproduction steps

Try passing prop as```renderNode``` and ```renderLabel``` to **CatalogGraphPage** it wont accept, though the child component can accept those props.

### 👀 Have you spent some time to check if this bug has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

### Are you willing to submit PR?

Yes I am willing to submit a PR!

**Comment by Rugvip** [comment]
Yep we can forward the props to the internal graph component from the page, just as was done in #16126 for the card 👍

**Comment by shubhamjajoo** [comment]
Closing this, as it's already merged !

**Comment by shubhamjajoo** [comment]
Created PR : https://github.com/backstage/backstage/pull/17097

**Comment by sennyeya** [comment]
@Rugvip Are all of the `initialState` props changeable? Might be out of scope of this request, but could see it being beneficial to move some `initialState` props to a `graphCardProps` props instead.

Current,
```
props: {
  initialState?: {
      selectedRelations?: string[];
      selectedKinds?: string[];
      rootEntityRefs?: string[];
      maxDepth?: number;
      unidirectional?: boolean;
      mergeRelations?: boolean;
      direction?: Direction;
      showFilters?: boolean;
      curve?: 'curveStepBefore' | 'curveMonotoneX';
    };
}
```

Proposed,
```
props: Partial<EntityRelationsGraphProps> & {
initialState?: {
    selectedRelations?: string[];
    selectedKinds?: string[];
  },
}
```

This would then remove the duplication 
```
              <EntityRelationsGraph
                rootEntityNames={rootEntityNames}
                maxDepth={maxDepth}
                ...
                mergeRelations={mergeRelations}
                unidirectional={unidirectional}
                onNodeClick={onNodeClick}
                direction={direction}
                relationPairs={relationPairs}
                zoom="enabled"
                curve={curve}
                />
```
with 
```
<EntityRelationsGraph
                ...
                {...props}
                kinds={
                  selectedKinds && selectedKinds.length > 0
                    ? selectedKinds
                    : undefined
                }
                relations={
                  selectedRelations && selectedRelations.length > 0
                    ? selectedRelations
                    : undefined
                }
                />
```

**Comment by shubhamjajoo** [comment]
@sennyeya , do you mean I should update this as part of this [PR](https://github.com/backstage/backstage/pull/17186)

**Comment by Rugvip** [comment]
Gonna close since this was fixed in #17186
